### PR TITLE
Final Capstone Project

### PR DESCRIPTION
**Pull request by MayPyone**
**### Dear code reviewers,**

Here are the implementations of our application.

- Each class was created in a separate .rb file.
- All associations (1-to-many relationships) and aggregations (parent-child relationships) between classes are implemented.
- All properties visible in the diagram are defined and set up in the constructor method except properties for the 1-to-many relationships
- All methods visible in the diagram are implemented add_item method() , can_be_archived?() and move_to_archive() in Item class.
- Book, MusicAlbum, and Game classes were inherited by Item class.
- Method overriding is applied.
- Unit tests were added.
- Created a main.rb file that will serve as our console app entry point.
- Our console app, at the start, will:

- [ ] Present the user with a list of options to perform.
- [ ] Let users choose an option.
- [ ] If needed, ask for parameters for the option.
- [ ] Have a way to quit the app.

- The following options should be available:

1. List all books
2. List all music albums
3. List all movies
4. List of games
5. List all genres (e.g 'Comedy', 'Thriller')
6. List all labels (e.g. 'Gift', 'New')
7. List all authors (e.g. 'Stephen King')
8. List all sources (e.g. 'From a friend', 'Online shop')
9. Add a book
10. Add a music album
11. Add a movie
12. Add a game

- All data are preserved by saving collections in .json files.
- Created a schema.sql file

### CODE REVIEW

**Comment by OLIPLICHE** [comment]
can you stop pushing while the review is going on? @MayPyone 
DONT COMMENT UNDER THIS POST, I will be deleting it

**Review by OLIPLICHE — CHANGES_REQUESTED** [review]
Hi Team, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- Recorded a video presenting the features of the project 👍 
- Each class was created in separated file 👍 
- All associations are implemented 👍 
- Add units test 👍 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by OLIPLICHE** [comment]
- [ ] **Hello Team. It is strongly prohibited to make the changes in the Pull request while your code is under Review**. @
- [ ] **NB: You have created The **linters mistakes** while I was submitting the review. I hope this will be fixed**.

File: spec/item_spec.rb [hunk 1]
```diff
@@ -0,0 +1,27 @@
+require './item'
+
+describe Item do
```
Comment (ridwanediallo) [inline spec/item_spec.rb:3]: - [ ] Good job adding tests to the `Item` class 👏.  However, I think there are some methods that are not tested yet: the `move_to_archive`,  the `genre`, the `author`  and the `label` methods. As per the requirement and as reminded by the previous reviewer, all methods inside the classes should be tested. So I would kindly suggest you test those four methods too.  🤗 👌

### Requirement 
Screenshot from 2023-11-03 14-59-27

File: spec/music_spec.rb [hunk 2]
```diff
@@ -1,37 +1,21 @@
...
-  def test_display_music_album_with_empty_list
-    assert_equal 'No music albums found', @music_module.display_music_album
-  end
-
-  def test_display_music_album_with_non_empty_list
-    @music_module.add_music_album('2023-11-04', 'y')
-    assert_equal "\nExisting music albums in the list:\n 1) Music Albem publish date: 2023-11-04 It is on spotify: true",
-                 @music_module.display_music_album
-  end
+describe 'Testing MusicAlbum class' do
```
Comment (ridwanediallo) [inline spec/music_spec.rb:3]: - [ ] As stated above, kindly let's ensure that the `can_be_archived?` inside the `MusicAlbum` class is tested too. 👌 😊

**Review by ridwanediallo — CHANGES_REQUESTED** [review]
# Changes Required ♻️
## Hi :wave:  team, 

AlmostThereReachGIF

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


## (Highlights) Good Points: :+1:
- You documented your work professionally. ✔
- You used GitHub flow correctly. ✔
- Your commit messages are well written. ✔


 ## Required Changes ♻️

- [ ] As you can see in the screenshot below, there are currently two failing tests. Please make sure to resolve the issues and ensure that all tests pass. 🚀 :+1: 

### Failling tests :point_down: 
Screenshot from 2023-11-04 12-54-18





## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/item_spec.rb [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+      item = Item.new("#{Date.today.year - 11}-01-01")
+      item.move_to_archive
+      expect(item.move_to_archive).to be true
+    end
+
+    it 'return false if the item cannot be archived' do
+      item = Item.new("#{Date.today.year - 9}-01-01")
+      expect(item.move_to_archive).to be false
+    end
+  end
```
Comment (prolajumokeoni) [inline spec/item_spec.rb:45]: - [ ] Kindly make sure that all tests added are passing, this test  is failng 👍🏿 

Screenshot_207

**Review by prolajumokeoni — CHANGES_REQUESTED** [review]
Hi @MayPyone 
Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

## Highlights
✔️ Testsed all methods requested by previus reviewer

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification @prolajumokeoni.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by hiromataba — APPROVED** [review]
# Status: Approved :tada: :tada: :tada: 
Hi Team   :wave: :wave: 

200w  

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**